src/pages/api/token.ts

```diff
@@ -0,0 +1,80 @@
+import jwt from "jsonwebtoken";
+import { NextApiResponse } from "next";
+import { Handler } from "next-iron-session";
+import { ExtendedRequest } from "src/@types/ExtendedRequest";
+import withSession from "src/middleware/withSession";
+
+function promisifiedJWTVerify(oauthData: {
+  clientSecret?: string;
+  accessToken: string;
+}): Promise<any> {
+  const { clientSecret, accessToken } = oauthData;
+
+  return new Promise((resolve, reject) => {
+    if (clientSecret) {
+      jwt.verify(
+        accessToken,
+        clientSecret,
+        async (error: jwt.VerifyErrors | null, decoded?: any) => {
+          resolve({ error, decoded });
+        },
```

Comment:
```suggestion
       (error: jwt.VerifyErrors | null, decoded?: any) => {
          resolve({ error, decoded });
        },
```

It'll work just as well :p